File: back-end/ShopHangTet/Models/Enums.cs
namespace ShopHangTet.Models;

public enum OrderStatus
{
    PAYMENT_CONFIRMING,
    PAYMENT_EXPIRED_INTERNAL,
    PREPARING,
    SHIPPING,
    PARTIAL_DELIVERY,
    DELIVERY_FAILED,
    COMPLETED,
    CANCELLED,
    REFUNDING,
    REFUNDED
}

public enum OrderType
{
    B2C,
    B2B
}

public enum OrderItemType
{
    READY_MADE,
    MIX_MATCH
}

/// Item Category Enum for Mix & Match validation
public enum ItemCategory
{
    DRINK,
    FOOD,
    NUT,
    ALCOHOL,
    SAVORY
}

public enum UserRole
{
    MEMBER,
    STAFF,
    ADMIN
}

public enum UserStatus
{
    ACTIVE,
    DISABLED
}

File: back-end-nodejs/src/models/enums.js
// ========== ENUMS ==========
export const OrderStatus = Object.freeze({
  PAYMENT_CONFIRMING: 'PAYMENT_CONFIRMING',
  PAYMENT_EXPIRED_INTERNAL: 'PAYMENT_EXPIRED_INTERNAL',
  PREPARING: 'PREPARING',
  SHIPPING: 'SHIPPING',
  PARTIAL_DELIVERY: 'PARTIAL_DELIVERY',
  DELIVERY_FAILED: 'DELIVERY_FAILED',
  COMPLETED: 'COMPLETED',
  CANCELLED: 'CANCELLED',
  REFUNDING: 'REFUNDING',
  REFUNDED: 'REFUNDED',
});

export const OrderType = Object.freeze({
  B2C: 'B2C',
  B2B: 'B2B',
});

export const OrderItemType = Object.freeze({
  READY_MADE: 'READY_MADE',
  MIX_MATCH: 'MIX_MATCH',
});

export const ItemCategory = Object.freeze({
  DRINK: 'DRINK',
  FOOD: 'FOOD',
  NUT: 'NUT',
  ALCOHOL: 'ALCOHOL',
});

export const UserRole = Object.freeze({
  MEMBER: 'MEMBER',
  STAFF: 'STAFF',
  ADMIN: 'ADMIN',
});

export const UserStatus = Object.freeze({
  ACTIVE: 'ACTIVE',
  DISABLED: 'DISABLED',
});
